src/wp-includes/pluggable.php

```diff
@@ -2840,6 +2837,13 @@ function get_avatar( $id_or_email, $size = 96, $default_value = '', $alt = '', $
 			$args['width'] = $args['size'];
 		}
 
+		/**
+		 * Update args with loading optimized attributes.
+		 *
+		 * @since 6.3
+		 */
+		$args = array_merge( $args, wp_get_loading_optimization_attributes( 'img', $args, 'get_avatar' ) );
```

Comment:
I had to modify the final args instead. Final image sizes were only available at this point. 

Let me know if this needs to be handled deferentially and also if i have to change the merge order.

cc: @felixarntz @joemcgill